pos-print/src/test/java/com/clouway/pos/print/adapter/http/ReportServiceTest.kt

```diff
@@ -43,96 +43,103 @@ class ReportServiceTest {
+    val reply = reportService.printOperatorReport(request.mockRequest(getOperatorReport()))
 
     assertThat(reply, isStatus(480))
     assertThat(reply, contains(ErrorResponse("Device can't connect.")))
   }
 
   @Test
   fun printReportForPeriod() {
-    val periodReportDTO = ReportService.PeriodReportDTO(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), "::any ip::", "SHORT")
-
     context.checking(object : Expectations() {
       init {
         oneOf(factory).getPrinter("::any ip::")
         will(AbstractExpectations.returnValue(printer))
-        oneOf(printer).reportForPeriod(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), PeriodType.SHORT)
+        oneOf(printer).reportForPeriod(LocalDateTime.parse("2017-01-01T00:00:00.000Z", DateTimeFormatter.ISO_DATE_TIME), LocalDateTime.parse("2017-01-02T00:00:00.000Z", DateTimeFormatter.ISO_DATE_TIME), PeriodType.SHORT)
         oneOf(printer).close()
       }
     })
 
-    val reply = reportService.printPeriodReport(request.mockRequest(periodReportDTO))
+    val reply = reportService.printPeriodReport(request.mockRequest(getPeriodReport()))
 
     assertThat(reply, isOk)
     assertThat(reply, contains("Report completed"))
   }
 
   @Test
   fun missingCashRegisterForPeriodReport() {
-    val periodReportDTO = ReportService.PeriodReportDTO(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), "::any ip::", "EXTENDED")
-
     context.checking(object : Expectations() {
       init {
         oneOf(factory).getPrinter("::any ip::")
         will(AbstractExpectations.throwException(DeviceNotFoundException()))
       }
     })
 
-    val reply = reportService.printPeriodReport(request.mockRequest(periodReportDTO))
+    val reply = reportService.printPeriodReport(request.mockRequest(getPeriodReport()))
 
     assertThat(reply, isBadRequest)
     assertThat(reply, contains(ErrorResponse("Device not found.")))
   }
 
   @Test
   fun connectionFailureForPeriodReport() {
-    val periodReportDTO = ReportService.PeriodReportDTO(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), "::any ip::", "EXTENDED")
-
     context.checking(object : Expectations() {
       init {
         oneOf(factory).getPrinter("::any ip::")
         will(AbstractExpectations.throwException(IOException()))
       }
     })
 
-    val reply = reportService.printPeriodReport(request.mockRequest(periodReportDTO))
+    val reply = reportService.printPeriodReport(request.mockRequest(getPeriodReport()))
```

Comment:
The following definition works on my side: 
```kotlin
 override fun <E> read(type: Class<E>): Request.RequestRead<E> {
        return object: Request.RequestRead<E> {
          override fun `as`(p0: Class<out Transport>): E {
            val transport = p0.newInstance()
            return transport.`in`(ByteArrayInputStream(e.toString().toByteArray()),type)
          }

        }
      }
```

This is how this is implemented in Java (bss)

```java
public <E> RequestRead<E> read(final Class<E> type) {
    return new RequestRead<E>() {
      @Override
      public E as(Class<? extends Transport> transport) {
          try {
            Transport transportInstance = FakeRequest.this.transport;
            if(FakeRequest.this.transport == null) {
              transportInstance = transport.newInstance();
            }
            return transportInstance.in(new ByteArrayInputStream(body.build().getBytes(StandardCharsets.UTF_8)), type);
          } catch (InstantiationException e) {
            e.printStackTrace();
          } catch (IllegalAccessException e) {
            e.printStackTrace();
          } catch (IOException e) {
            e.printStackTrace();
          }
        return null;
      }
    };
  }
```

As you can see both versions are using the passed transport.